packages/shopping-cart/src/use-shopping-cart.ts

```
@@ -1,11 +1,49 @@
-import { useContext } from 'react';
+import debugFactory from 'debug';
+import { useContext, useState, useEffect, useRef, useMemo } from 'react';
 import ShoppingCartContext from './shopping-cart-context';
-import type { ShoppingCartManager } from './types';
+import ShoppingCartOptionsContext from './shopping-cart-options-context';
+import useRefetchOnFocus from './use-refetch-on-focus';
+import type { UseShoppingCart } from './types';
 
-export default function useShoppingCart(): ShoppingCartManager {
-	const manager = useContext( ShoppingCartContext );
-	if ( ! manager ) {
+const debug = debugFactory( 'shopping-cart:use-shopping-cart' );
+
+export default function useShoppingCart( cartKey?: string | undefined ): UseShoppingCart {
+	const managerClient = useContext( ShoppingCartContext );
+	if ( ! managerClient ) {
 		throw new Error( 'useShoppingCart must be used inside a ShoppingCartProvider' );
 	}
-	return manager;
+
+	const { defaultCartKey } = useContext( ShoppingCartOptionsContext ) ?? {};
+	const finalCartKey = cartKey ?? defaultCartKey;
+	debug( `getting cart manager for cartKey ${ finalCartKey }` );
+	const manager = managerClient.forCartKey( finalCartKey );
```

What if both `cartKey` and `defaultCartKey` are `undefined`? Should we report that as error?

---

That was intentional. If for example you don't know the cart key yet but you have to call the hook anyway, it's essentially a noop (it returns an empty cart that is always marked as loading).